ios/MagnetScannerReact.m

```diff
@@ -0,0 +1,13 @@
+#import "RCTBridgeModule.h"
+
+@interface RCT_EXTERN_MODULE(MagnetScannerReact, NSObject);
+
+RCT_EXTERN_METHOD(
+                  start:(RCTPromiseResolveBlock)resolve
+                  reject:(RCTPromiseRejectBlock)reject)
+
+RCT_EXTERN_METHOD(
+                  stop:(RCTPromiseResolveBlock)resolve
+                  reject:(RCTPromiseRejectBlock)reject)
+
+@end
```

Comment:
Both `start()` and `stop()` return promises now as we have given native code responsibility to manage device dependencies before scanner can run. For example prompting the user for bluetooth is a slow async process which blocks the magnet-scanner-android from running. The promise doesn't resolve until these permission dependencies have been checked and prompted.